### PR TITLE
adding details on how to config the HDFS FW

### PR DESCRIPTION
**Pull request by kensipe**

### CODE REVIEW

**Comment by kensipe** [comment]
@gabrielhartmann it would be nice to get a review on these docs

File: README.md [hunk 1]
```diff
@@ -28,11 +28,8 @@ Installing HDFS-Mesos on your Cluster
 3. Optional: Customize any additional configurations that weren't updated at compile time in `hdfs-mesos-*/etc/hadoop/*-site.xml` Note that if you update hdfs-site.xml, it will be used by the scheduler and bundled with the executors. However, core-site.xml and mesos-site.xml will be used by the scheduler only.
 4. Check that `hostname` on that node resolves to a non-localhost IP; update /etc/hosts if necessary.
 
-### If you have Hadoop pre-installed in your cluster
-If you have Hadoop installed across your cluster, you don't need the Mesos scheduler application to distribute the binaries. You can set the `mesos.hdfs.native-hadoop-binaries` configuration parameter in `mesos-site.xml` if don't want the binaries distributed.
```
Comment (gabrielhartmann) [inline README.md]: You skipped a word: "if YOU don't the binaries"

File: README.md [hunk 2]
```diff
@@ -28,11 +28,8 @@ Installing HDFS-Mesos on your Cluster
...
 
-### Mesos-DNS custom configuration
-You can see the example configuration in the `example-conf/dcos` directory. Since Mesos-DNS provides native bindings for master detection, we can simply use those names in our mesos and hdfs configurations. The example configuration assumes your Mesos masters and your zookeeper nodes are colocated. If they aren't you'll need to specify your zookeeper nodes separately. Also, note that you are using the example in `example-conf/dcos`, the `mesos.hdfs.native-hadoop-binaries` property needs to be set to `false` if your HDFS binaries are not predistributed.
```
Comment (gabrielhartmann) [inline README.md]: "You can see AN example configuration ..."

"Also, note that IF you are not using ..."

File: config.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+## Configuration of HDFS framework and HDFS
+
+The configuration of HDFS and this framework are managed via:
+
+* hdfs-site.xml
+* mesos-site.xml
+* system env vars and proeprties
```
Comment (gabrielhartmann) [inline config.md]: properties

File: config.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+The hdfs-site.xml file is used to configure the hdfs cluster.  The values must match the configuration fo the scheduler.  For this
```
Comment (gabrielhartmann) [inline config.md]: "OF the scheduler"

File: config.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+The hdfs-site.xml file is used to configure the hdfs cluster.  The values must match the configuration fo the scheduler.  For this 
+reason the hdfs-site.xml is generally "fetched" or refreshed from the scheduler when a node is started.   The normal configuration of
+the hdfs-site.xml has variables which are replaced by the scheduler when the xml file is fetched by the node.  An example of these
+variables is `${frameworkName}`.   The scheduler code that does the variable replacement is handled by ConfigServer.java.  An
+example of this variable replacement is `model.put("frameworkName", hdfsFrameworkConfig.getFrameworkName());`
+
+For environments which are "provisioned" with hdfs and managed by hdfs-mesos it is expected that the values of this xml file are 
+established for the deployment.   Environments which are designated as `mesos.hdfs.native-hadoop-binaries` == true in the `mesos-site.xml`,
+there is no refresh of the `hdfs-site.xml` file.
```
Comment (gabrielhartmann) [inline config.md]: I don't understand this sentence, "Environments which are designated..."

File: config.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+* mesos.hdfs.framework.mnt.path - If native-hadoop-binaries == false, this is the location a symlink will be provided to execute hdfs commands on the slave.  The default is "/opt/mesosphere"
+* mesos.hdfs.state.zk - The zookeeper that the scheduler will use to store state.  The default is "localhost:2181"
+* mesos.master.uri - The zookeeper or mesos-master url that will be used to discover the mesos-master for scheduler registration. The default is "localhost:2181"
+* mesos.hdfs.zkfc.ha.zookeeper.quorum - The zookeeper that HDFS (not the framework) will use for HA mode.  The default is "localhost:2181"
+
+There are additional configurations for executor jvm and resource management of the nodes.
+
+## System Environment Variables
+
+All of configuration flags previous defined can be override with system environment variables.  The format to use to over a variable is to
```
Comment (gabrielhartmann) [inline config.md]: "All of the configuration flags previously defined can be overridden by system environment variables."

File: config.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+upper case the string and replace dots (".") with underscores ("_").  so to override the `mesos.hdfs.framework.name`, the value is `MESOS_HDFS_FRAMEWORK_NAME=unicorn".
```
Comment (gabrielhartmann) [inline config.md]: "The format to use to OVERRIDE a variable is to ..."

File: config.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+To use this value, export the value, then start the scheduler.  If a value is overridden by the system environment variable it will be propagated to
+the executors.
+
+## Custom Configurations
+
+### Mesos-DNS custom configuration
+You can see the example configuration in the `example-conf/dcos` directory. Since Mesos-DNS provides native bindings for master detection, we can simply use those names in our mesos and hdfs configurations. The example configuration assumes your Mesos masters and your zookeeper nodes are colocated. If they aren't you'll need to specify your zookeeper nodes separately. Also, note that you are using the example in `example-conf/dcos`, the `mesos.hdfs.native-hadoop-binaries` property needs to be set to `false` if your HDFS binaries are not predistributed.
```
Comment (gabrielhartmann) [inline config.md]: This is copy pasted from elsewhere.  There was a typo in the other version, so fix it here too.

**Comment by gabrielhartmann** [comment]
Minor copy editing stuff.  The only thing I'm not crazy about is the use of a string format convention to indicate overrides.  Could we have a config flag, that indicates we are using going to use overrides and then go look for environment variables in the format you described?  I just prefer being explicit about the override behavior as it's potentially surprising.

No comments are blocking.  LGTM.

**Comment by kensipe** [comment]
thanks!  pertaining to your last comment... I want to completely remove the meso-site.xml.  All configurations should be standardized across all of our frameworks.  The common approach for most is to use system env vars are args.  This is using env vars.     When we remove mesos-site.xml then technically it will not be an override of an xml, it will be an override of the default which would be expected behavior for anyone that uses an env var.

File: README.md (hunk 1 above)
Comment (kensipe) [inline README.md]: that was text previous to me... and this review is on text that was removed... or pasted in another place... 
I'll fix it there

File: README.md (hunk 2 above)
Comment (kensipe) [inline README.md]: same here

**Comment by elingg** [comment]
Completely agree with removing mesos-site.xml although it will be a significant change. Nice work @kensipe and @gabrielhartmann

**Pull request merged**